src/System.Data.SqlClient/tests/ManualTests/SQL/ConnectionPoolTest/PoolBlockPeriodTest.cs

```diff
@@ -0,0 +1,344 @@
+
+                Assert.True(CompareWithMargin(timeInSecs, blockingPeriodInSeconds, comparisonMargin),
+                    $"#{counter} blocking Period must be {blockingPeriodInSeconds} seconds");
+
+                counter++;
+                blockingPeriodInSeconds = blockingPeriodInSeconds * 2;
+            }
+
+            //Test after max period
+            for (int i = 0; i < 2; i++)
+            {
+                var timeInSecs = GetBlockingPeriodTimeInSeconds(connString, setting);
+                Debug.WriteLine($"#{counter} Connection.Open (After Max) executed in {timeInSecs} seconds");
+                Assert.True(CompareWithMargin(timeInSecs, _maxBlockingPeriodInSeconds, comparisonMargin),
+                     $"#{counter} blocking Period must be capped at max blocking period '{_maxBlockingPeriodInSeconds} seconds'");
+                counter++;
+            }
+
+        }
+
+        [Theory]
+        [InlineData("Azure with Default Policy must Disable blocking (.database.windows.net)", new object[] { "nonexistance.database.windows.net" })]
+        [InlineData("Azure with Default Policy must Disable blocking (.database.chinacloudapi.cn)", new object[] { "nonexistance.database.chinacloudapi.cn" })]
+        [InlineData("Azure with Default Policy must Disable blocking (.database.usgovcloudapi.net)", new object[] { "nonexistance.database.usgovcloudapi.net" })]
+        [InlineData("Azure with Default Policy must Disable blocking (.database.cloudapi.de)", new object[] { "nonexistance.database.cloudapi.de" })]
+        [InlineData("Azure with Default Policy must Disable blocking (MIXED CASES) (.database.WINDOWS.net)", new object[] { "nonexistance.database.WINDOWS.net" })]
+        [InlineData("Azure with Default Policy must Disable blocking (PORT) (.database.WINDOWS.net,1234)", new object[] { "nonexistance.database.WINDOWS.net,1234" })]
+        [InlineData("Azure with Default Policy must Disable blocking (INSTANCE NAME) (.database.WINDOWS.net,1234\\INSTANCE_NAME)", new object[] { "nonexistance.database.WINDOWS.net,1234\\INSTANCE_NAME" })]
+        [InlineData("Azure with Auto Policy must Disable Blocking", new object[] { "nonexistance.database.windows.net", PoolBlockingPeriod.Auto })]
+        [InlineData("Azure with Always Policy must Enable Blocking", new object[] { "nonexistance.database.windows.net", PoolBlockingPeriod.AlwaysBlock })]
+        [InlineData("Azure with Never Policy must Disable Blocking", new object[] { "nonexistance.database.windows.net", PoolBlockingPeriod.NeverBlock })]
+        public void TestAzureBlockingPeriod(string description, object[] Params)
+        {
+            var serverName = Params[0] as string;
+            PoolBlockingPeriod? policy = null;
+            if (Params.Length > 1)
+            {
+                policy = (PoolBlockingPeriod)Params[1];
+            }
+
+            var connString = CreateConnectionString(serverName, policy);
```

Comment:
Nit: all of these vars should be replaced with the actual type

Reply:
done